server.js

```diff
@@ -24,6 +24,61 @@ app.get("/", (req, res) => {
   res.send("Hello Technigo!");
 });
 
+app.get("/books", (req, res) => {
+  const { title, authors } = req.query;
+  let allBooks = books;
+/*  Write for example: http://localhost:8080/books?authors=Gary Paulsen to get all the authors books
+    Write for example: http://localhost:8080/books?title=Hatchet to get the singel books info */
+
+  if (authors) {
+    allBooks = allBooks.filter(singleBook => singleBook.authors.toLowerCase() === authors.toLowerCase());
+  }
+
+  if (title) {
+    allBooks = allBooks.filter(singleBook => singleBook.title.toLowerCase() === title.toLowerCase());
+  } 
+
+  res.status(200).json({
+    success: true,
+    message: "OK",
+    body: { 
+      books: allBooks
+    }
+    });
+});
+
+app.get("/books/:id", (req, res) => {
+  const singleBook = books.find((book) => {
+    return book.bookID === Number(req.params.id)
```

Comment:
This solution with Number is very interesting. I like how you use this way of achieving the desired data